### PR TITLE
Back-End Environment

### PR DESCRIPTION
**Pull request by ngowler-rrc**
Nick's Back-End Environment assignment.

### CODE REVIEW

File: src/portfolio/portfolioPerformance.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+export interface PortfolioPerformance {
+    initialInvestment: number;
+    currentValue: number;
+    profitOrLoss: number;
+    percentageChange: number;
+    performanceSummary: string;
+}
+
+export function calculatePortfolioPerformance(initialInvestment: number, currentValue: number): PortfolioPerformance {
```
Comment (DaveRRC) [inline src/portfolio/portfolioPerformance.ts:9]: You can use docs for your functions: https://tsdoc.org

File: src/portfolio/portfolioPerformance.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	const portfolioPerformance: PortfolioPerformance = {
+		initialInvestment,
+		currentValue,
+		profitOrLoss,
+		percentageChange,
+		performanceSummary
+	}
+
+    return portfolioPerformance;
```
Comment (DaveRRC) [inline src/portfolio/portfolioPerformance.ts:53]: indentation gets a little wonky in this function

File: src/portfolio/portfolioPerformance.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+export function findLargestHolding(assets: Asset[]): Asset {
+    let largestHolding: Asset = assets[0];
+
+    for (let i: number = 1; i < assets.length; i++) {
+        if (assets[i].value > largestHolding.value) {
+            largestHolding = assets[i];
+        }
+    }
```
Comment (DaveRRC) [inline src/portfolio/portfolioPerformance.ts:68]: This totally works, you could also try out [Array sorts](https://www.w3schools.com/js/js_array_sort.asp#mark_objects:~:text=Sorting%20Object%20Arrays)

File: test/portfolioPreformance.test.ts
Comment (DaveRRC) [inline test/portfolioPreformance.test.ts:1]: Some points where indentation goes off in this file

File: src/portfolio/portfolioPerformance.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    assets.forEach(asset => {
+        totalValue += asset.value;
+    });
+    assets.forEach(asset => {
+        const percentage: number = (asset.value / totalValue) * 100;
+        assetAllocationPercentages.push({ name: asset.name, percentage });
+    });
+
+    return assetAllocationPercentages;
+}
```
Comment (DaveRRC) [inline src/portfolio/portfolioPerformance.ts:91]: What would happen if `assets` were empty when passed to either function?

**Review by DaveRRC — APPROVED** [review]
Overall very happy with your work, just be sure to add function documentation in the future. I'd also recommend testing out edge cases like passing an empty array to a function and seeing what happens.
